com-api/examples/basic-consumer-producer.rs

```diff
@@ -12,23 +12,23 @@
 use com_api::*;
 use com_api_gen::*;
 
-
 // Example struct demonstrating composition with VehicleConsumer
 pub struct VehicleMonitor<R: Runtime> {
     consumer: VehicleConsumer<R>,
+    producer: VehicleOfferedProducer<R>,
```

Comment:
```suggestion
    producer: VehicleOfferedProducer<R>,
    tire_subscriber: <<R as Runtime>::Subscriber<Tire> as Subscriber<Tire, R>>::Subscription,
```
Subscriber should be created before sending data to make sure those wont be dropped